### PR TITLE
aws_transit_gateway_multicast_group_source

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

It contains all the singular and plural resources of the above resource. No integration tests with terraform are done as there is no terraform resource available.

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [ ] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Review by sathish-progress — APPROVED** [review]
LGTM, but please update readme

File: docs/resources/aws_transit_gateway_multicast_group_sources.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+`transit_gateway_multicast_domain_id` _(required)_
+
+The transit_gateway_multicast_domain_id is the ID of the multicast domain transit gateway.
+
+For additional information, see the [AWS documentation on AWS EC2 TransitGatewayMulticastGroupSource](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-transitgatewaymulticastgroupsource.html).
+
+## Properties
+
+| Property | Description|
```
Comment (sathish-progress) [inline docs/resources/aws_transit_gateway_multicast_group_sources.md:30]: if possible add fields

Comment (soumyo13) [reply docs/resources/aws_transit_gateway_multicast_group_sources.md]: Done

File: docs/resources/aws_transit_gateway_multicast_group_source.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+## Properties
+
+| Property | Description|
+| --- | --- |
+| group_ip_address | The IP address of the transit gateway multicast group. |
+| transit_gateway_attachment_id | The attachment id of the transit gateway. |
+| subnet_id | The ID of the subnet. |
+| resource_id | The ID of the resource. |
+| resource_type | The type of the resource. |
+| resource_owner_id | The Owner ID of the resource. |
```
Comment (IanMadd) [inline docs/resources/aws_transit_gateway_multicast_group_source.md:37]: Should `resource_owner_id` be in here? Also, I don't see `TransitGatewayMulticastDomainId` in the properties.

Comment (soumyo13) [reply docs/resources/aws_transit_gateway_multicast_group_source.md:37]: It should be resource_owner_id.
And as this API don't have a TransitGatewayMulticastDomainId property so it is not mentioned over here.

If the user wants to check whether the TransitGatewayMulticastDomainId is present or not then he should use the example as 

describe aws_transit_gateway_multicast_group_source(transit_gateway_multicast_domain_id: "tgw-mcast-domain-1234567890") do
      it { should exist }
    end

Please see the docs https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/EC2/Client.html#search_transit_gateway_multicast_groups-instance_method

**Review by IanMadd — APPROVED** [review]

**Pull request merged**